src/cpp/src/model_runner.hpp

```diff
@@ -49,7 +70,7 @@ class ModelRunner {
             // PA specific parameters
             past_lens(ov::element::i32, {batch_size_in_sequences}),
             subsequence_begins(ov::element::i32, {batch_size_in_sequences + 1}),
-            block_indices(ov::element::i32, {total_num_blocks}),
+            // block_indices are handled in a special fashion below
```

Comment:
do we need a special handling here? 
Looks like because we have changed scheduler output from a single block table to per-layer block tables, we just need to copy num_layersX more data in the same fashion as before.

Reply:
That's correct right now, but @l-bat is currently evaluating cache eviction strategies where each layer could have a different number of blocks occupied, so looking forward we could expect special handling of block indices in the future. I can merge the loop back for the current state, if you insist.